### ISSUE TITLE
Sequential timesMat_local segfaults when GPU support in enabled

### ISSUE THREAD

**Issue by pelesh** [LLNL/hiop#210]
There is a segmentation fault in `hiopMatrixDenseRowMajor::timesMat_local`. The backtrace indicates that HiOp links to NVBlas instead of host implementation of BLAS:
```shell
#0  0x00007fffc9d3262d in vfprintf () from /usr/lib64/libc.so.6
#1  0x00007fffc9d3d407 in fprintf () from /usr/lib64/libc.so.6
#2  0x00007fffd20d9266 in dgemm_ () from /share/apps/cuda/11.1/lib64/libnvblas.so.11
#3  0x0000000000421209 in hiop::hiopMatrixDenseRowMajor::timesMat_local (this=0xca8830, beta=<optimized out>, 
    W_=..., alpha=<optimized out>, X_=...)
    at /people/pele118/src/hiop/hiop/src/LinAlg/hiopMatrixDenseRowMajor.cpp:549
#4  0x0000000000414db6 in hiop::tests::MatrixTestsDense::matrixTimesMat (this=this@entry=0x7fffffffc558, A=..., 
    X=..., W=..., rank=rank@entry=0) at /people/pele118/src/hiop/hiop/tests/LinAlg/matrixTestsDense.hpp:266
#5  0x0000000000412c22 in runTests<hiop::tests::MatrixTestsDenseRowMajor> (comm=<optimized out>, 
    mem_space=0x4b71fb "default") at /people/pele118/src/hiop/hiop/tests/testMatrixDense.cpp:230
#6  0x0000000000410686 in main (argc=<optimized out>, argv=<optimized out>)
    at /people/pele118/src/hiop/hiop/tests/testMatrixDense.cpp:102
```
This was reproduced from `develop` branch with following modules loaded:
```shell
$ module list
Currently Loaded Modulefiles:
  1) gcc/7.3.0       2) cmake/3.15.3    3) openmpi/3.1.3   4) metis/5.1.0     5) cuda/11.1
```

CC @ashermancinelli

**Comment by cnpetra** [comment]
@jwang125 : didn't you run into something like this on lassen?

**Comment by pelesh** [comment]
Another piece of info: This happens on Intel platform (Broadwell), but not on Power9. This is why this went under the radar. This could be specific to BLAS and CMake configurations on the intel machine. I'll investigate more.

@cnpetra, do we need NVBlas as Magma dependency or you use it somewhere else in the code?

**Comment by cnpetra** [comment]
nvblas is a dependency of magma. I suspect it somehow replaces the cpu blas library during linking phase (likely cmake-related as you said) and then it fails when hiop makes the usual calls to blas...

**Comment by ashermancinelli** [comment]
[From the NVBLAS docs:](https://docs.nvidia.com/cuda/nvblas/index.html)

> To use the NVBLAS Library, the user application must be relinked against NVBLAS in addition to the original CPU Blas (technically only NVBLAS is needed unless some BLAS routines not supported by NVBLAS are used by the application). To be sure that the linker links against the exposed symbols of NVBLAS and not the ones from the CPU Blas, the NVBLAS Library needs to be put before the CPU Blas on the linkage command line.

It looks like using host and nvblas in the same application might be difficult.

**Comment by jwang125** [comment]
> @jwang125 : didn't you run into something like this on lassen?

I don't remember this error, sorry.

**Comment by cnpetra** [comment]
closed by PR#213
